File: IncidentSystem/src/main/java/com/example/IncidentSystem/Models/Employee.java
package com.example.IncidentSystem.Models;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.Table;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

@Entity
@Table(name = "Employee")
public class Employee {

	@Id
	@Column
	private int id;
	
	@Column
	private String name;
	
	@ManyToOne(fetch = FetchType.EAGER,  cascade=CascadeType.ALL)
	@JoinColumn(name = "team_id")
	private Team team;
	
	@ManyToOne(fetch = FetchType.EAGER,  cascade=CascadeType.ALL)
	@JoinColumn(name = "system_role_id")
	private SystemRole system_role;
	
	@ManyToOne(fetch = FetchType.EAGER,  cascade=CascadeType.ALL)
	@JoinColumn(name = "team_id")
	private BusinessRole business_role;
}

File: IncidentSystemWeb/src/models/Employee.ts
import { BusinessRole } from "./BusinessRole";
import { Incident } from "./Incident";
import { Model } from "./Model";
import { SystemRole } from "./SystemRole";
import { Team } from "./Team";

export interface Employee extends Model{
    name:string;
    team:Team;
    system_role:SystemRole;
    business_role:BusinessRole;
    requests:Incident[];
    attributed_incidents:Incident[];
}